_posts/2019-06-04-loading-large-image-data-with-dask-array.md

```diff
@@ -67,160 +173,923 @@ pipeline.
-become prohibitive once we try to load more data than we have memory. So we
-might use `delayed` to replace this with something like this. We can sub in the
-shape and type from a single file that we have inspected. This could be used
-for multiple files if we know them to be uniform and of the same type. This
-gives us a Dask Array that represents the data on disk for us without using up
-our memory to store it. Pulling from the data on demand.
+Each filename corresponds to some 3d chunk of a larger image.  We can look at a
+few 2d slices of this single 3d chunk to get some context.
 
 ```python
-import dask.array as da
-from dask.delayed import delayed
-from skimage.io import imread
-a = da.from_delayed(delayed(imread)("mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif"),
-                    <shape>,
-                    <dtype>)
+import matplotlib.pyplot as plt
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
+```
+
+<img src="https://raw.githubusercontent.com/mrocklin/raw-host/gh-pages/images/aollsm-sample-1.png"
+     width="60%">
+
+```python
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
 ```
 
-If we want to simplify this a bit, we can use a pre-rolled implementation that
-does this work for us like [dask-image's `imread`](
-https://dask-image.readthedocs.io/en/latest/dask_image.imread.html ). This
-combines the simplicity of an API like scikit-image's with the benefits of
-leveraging Dask to manage the larger data.
+<img src="https://raw.githubusercontent.com/mrocklin/raw-host/gh-pages/images/aollsm-sample-2.png"
+     width="60%">
 
 ```python
-from dask_image.imread import imread
-a = imread("mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif")
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
 ```
 
-If we want to handle loading some other kind of data, we can take our own
-loading function and combine it with `delayed`. This way we can handle
-operations like reading from a database or requesting data from the cloud.
+<img src="https://raw.githubusercontent.com/mrocklin/raw-host/gh-pages/images/aollsm-sample-3.png"
+     width="60%">
 
-Combining data loaded with Dask
--------------------------------
 
-Thus far we have only showed how one might load one image file. However we
-generally have many such files. At a first pass we might try to load up the
-collection of files in a list.
+### Investigate Filename Structure
+
+This are slices from only one chunk of a much larger aggregate image.
+Our main task in dealing with large image stacks is how to arrange the pieces
+back together.
+It is common to see a naming structure in the filenames. Each
+filename then may indicate a channel, time step, and spatial location with the
+`<i>` being some numeric values (possibly with units). Individual filenames may
+have more or less information and may notate it differently than we have.
+
+```
+mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif  # TODO: use filename structure from the AOLLSM example
```

Comment:
I wouldn't get too focused on using the exact file name structure as AOLLSM. Ultimately I think this will distract from the point. Particularly as some parameters end up being repeats of previous ones (like camera, channel, and wavelength). The main thing we should aim to get across here is that there are a few common parameters that may show up in the filename and here is how we use them to construct the larger representation of the data on disk.